src/App.tsx

```diff
@@ -1,14 +1,38 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
-export class App extends React.Component<{}, State> {
-  state: State = {};
+export class App extends React.PureComponent<{}, State> {
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    const key = name as keyof State;
+
+    this.setState({ [key]: value });
+  };
+
+  getVisibleMovies = (movies: Movie[]) => {
+    const { query } = this.state;
+
+    return [...movies].filter(movie => (
```

Comment:
filter не меняет оригинальный массив
```suggestion
    return movies.filter(movie => (
```